### ISSUE TITLE
Pages -> View does not work in 2.3.0

### ISSUE THREAD

**Issue by beltran-rubo** [magento/magento2#19685]
### Preconditions (*)
1. Install Magento via CLI using the "developer" mode
2. Reindex & flush cache

### Steps to reproduce (*)
1. Magento 2.3.0
2. Go to Pages -> Select one -> Click on View

### Expected result (*)

1. Show the Magento page in the browser

### Actual result (*)
```
1 exception(s):
Exception #0 (Magento\Framework\Exception\NoSuchEntityException): No such entity with id = 0

Exception #0 (Magento\Framework\Exception\NoSuchEntityException): No such entity with id = 0
#0 /htdocs/vendor/magento/module-catalog/Model/CategoryRepository.php(141): Magento\Framework\Exception\NoSuchEntityException::singleField('id', '0')
#1 /htdocs/vendor/magento/module-catalog/Model/Layer.php(202): Magento\Catalog\Model\CategoryRepository->get('0')
#2 
/htdocs/generated/code/Magento/Catalog/Model/Layer/Category/Interceptor.php(76): Magento\Catalog\Model\Layer->getCurrentCategory()
#3 /htdocs/vendor/magento/module-catalog/Plugin/Block/Topmenu.php(146): Magento\Catalog\Model\Layer\Category\Interceptor->getCurrentCategory()
#4 /htdocs/vendor/magento/module-catalog/Plugin/Block/Topmenu.php(79): Magento\Catalog\Plugin\Block\Topmenu->getCurrentCategory()
#5 /htdocs/vendor/magento/framework/Interception/Interceptor.php(121): Magento\Catalog\Plugin\Block\Topmenu->beforeGetHtml(Object(Magento\Theme\Block\Html\Topmenu\Interceptor), 'level-top', 'submenu', 0)
...
```

**Comment by magento-engcom-team** [comment]
Hi @beltran-rubo. Thank you for your report. 
To help us process this issue please make sure that you provided the following information:
- [ ] Summary of the issue
- [ ] Information on your environment
- [ ] Steps to reproduce
- [ ] Expected and actual results

Please make sure that the issue is reproducible on the vanilla Magento instance following **Steps to reproduce**. To deploy vanilla Magento instance on our environment, please, add a comment to the issue: 

`@magento-engcom-team give me $VERSION instance` 

where `$VERSION` is [version tags](https://github.com/magento/magento2/tags) (starting from 2.2.0+) or develop branches (for example: 2.3-develop). 
For more details, please, review the [Magento Contributor Assistant](https://devdocs.magento.com/guides/v2.2/contributor-guide/contributing.html#contributor-assist) documentation.

@beltran-rubo do you confirm that you was able to reproduce the issue on vanilla Magento instance following steps to reproduce?
- [ ] yes
- [ ] no

**Comment by beltran-rubo** [comment]
@magento-engcom-team give me 2.3 instance

**Comment by magento-engcom-team** [comment]
Hi @engcom-backlog-nazar. Thank you for working on this issue. 
In order to make sure that issue has enough information and ready for development, please read and check the following instruction: :point_down:
- [x] 1. Verify that issue has all the required information. (**Preconditions**, **Steps to reproduce**, **Expected result**, **Actual result**).<details><summary>Details</summary>If the issue has a valid description, the label `Issue: Format is valid` will be added to the issue automatically. Please, edit issue description if needed, until label `Issue: Format is valid` appears.</details>

- [x] 2. Verify that issue has a meaningful description and provides enough information to reproduce the issue. If the report is valid, add `Issue: Clear Description` label to the issue by yourself.

- [x] 3. Add `Component: XXXXX` label(s) to the ticket, indicating the components it may be related to.
- [x] 4. Verify that the issue is reproducible on `2.3-develop` branch<details><summary>Details</summary>- Add the comment `@magento-engcom-team give me 2.3-develop instance` to deploy test instance on Magento infrastructure. <br/> - If the issue is reproducible on `2.3-develop` branch, please, add the label `Reproduced on 2.3.x`.<br/>- If the issue is not reproducible, add your comment that issue is not reproducible and close the issue and **_stop verification process here_**!
</details>

- [ ] 5. Verify that the issue is reproducible on `2.2-develop` branch. <details><summary>Details</summary>- Add the comment `@magento-engcom-team give me 2.2-develop instance` to deploy test instance on Magento infrastructure. <br/>- If the issue is reproducible on `2.2-develop` branch, please add the label `Reproduced on 2.2.x`</details>

- [x] 6. Add label `Issue: Confirmed` once verification is complete. 

- [x] 7. Make sure that automatic system confirms that report has been added to the backlog.

**Comment by ghost** [comment]
Hi @beltran-rubo  i'm not able to reproduce following steps your described, only 2.2.0+ bug reports are considered nowadays. Try upgrading.

**Comment by beltran-rubo** [comment]
@engcom-backlog-nazar sorry, the version was wrong in the title. This happen in the new 2.3.0 (and not in the previous versions)

**Comment by ghost** [comment]
@beltran-rubo ok, i'm will retest

**Comment by magento-engcom-team** [comment]
@engcom-backlog-nazar Thank you for verifying the issue. Based on the provided information internal tickets `MAGETWO-97038` were created

**Comment by kunj1988** [comment]
I just tried this and working fine for me.

**Comment by magento-engcom-team** [comment]
Hi @eduard13. Thank you for working on this issue. 
Looks like this issue is already verified and confirmed. But if your want to validate it one more time, please, go though the following instruction:
- [x] 1. Add/Edit `Component: XXXXX` label(s) to the ticket, indicating the components it may be related to.

- [x] 2. Verify that the issue is reproducible on `2.3-develop` branch<details><summary>Details</summary>- Add the comment `@magento-engcom-team give me 2.3-develop instance` to deploy test instance on Magento infrastructure. <br/> - If the issue is reproducible on `2.3-develop` branch, please, add the label `Reproduced on 2.3.x`.<br/>- If the issue is not reproducible, add your comment that issue is not reproducible and close the issue and **_stop verification process here_**!
</details>

- [ ] 3. Verify that the issue is reproducible on `2.2-develop` branch. <details><summary>Details</summary>- Add the comment `@magento-engcom-team give me 2.2-develop instance` to deploy test instance on Magento infrastructure. <br/>- If the issue is reproducible on `2.2-develop` branch, please add the label `Reproduced on 2.2.x`</details>

- [ ] 4. If the issue is not relevant or is not reproducible any more, feel free to close it.

**Comment by nathanjosiah** [comment]
This is a duplicate of internal issue `MC-5461`. This issue will be fixed in 2.3.1

**Comment by eduard13** [comment]
@nathanjosiah thank you for letting me know, I won't continue with this one then.

**Comment by nathanjosiah** [comment]
No problem, we have a fix that should be merged into 2.3-develop within the next week

**Comment by jignesh-baldha** [comment]
@nathanjosiah I've got a fix which solved this issue, May I create PR now?

**Comment by magento-engcom-team** [comment]
Hi @jignesh-baldha. Thank you for working on this issue. 
Looks like this issue is already verified and confirmed. But if your want to validate it one more time, please, go though the following instruction:
- [x] 1. Add/Edit `Component: XXXXX` label(s) to the ticket, indicating the components it may be related to.

- [x] 2. Verify that the issue is reproducible on `2.3-develop` branch<details><summary>Details</summary>- Add the comment `@magento-engcom-team give me 2.3-develop instance` to deploy test instance on Magento infrastructure. <br/> - If the issue is reproducible on `2.3-develop` branch, please, add the label `Reproduced on 2.3.x`.<br/>- If the issue is not reproducible, add your comment that issue is not reproducible and close the issue and **_stop verification process here_**!
</details>

- [x] 3. Verify that the issue is reproducible on `2.2-develop` branch. <details><summary>Details</summary>- Add the comment `@magento-engcom-team give me 2.2-develop instance` to deploy test instance on Magento infrastructure. <br/>- If the issue is reproducible on `2.2-develop` branch, please add the label `Reproduced on 2.2.x`</details>

- [ ] 4. If the issue is not relevant or is not reproducible any more, feel free to close it.

**Comment by nathanjosiah** [comment]
@jignesh-baldha Thank you for investigating this issue. We have a fix internally that will be merged into 2.3-develop within the next few business days.

**Comment by sdzhepa** [comment]
Hello @beltran-rubo @eduard13 @jignesh-baldha 

This issue was fixed in the scope of internal ticket `MC-5461`
The fix should be available for 2.3.1 (curently in 2.3-develop)
Please, see commits for more details:
- d6f7d244fffade4764afb53fc0d4b932f5b180a1
- 25473b75ed223ed37cd677dfa34cc3ac4946bd2a

This ticket can be closed.
Please, feel free to reopen or create a new one if issue still exists or was not fully fixed

Thank you for feedback and collaboration

**Comment by AsadKhan95** [comment]
I'm getting the same issue in M 2.2.10. While when I add new page then it gives me the above error while when I apply the page then it gives me nothing in the frontend.
https://magento.stackexchange.com/questions/292523/adding-new-page-in-admin-giving-error-magento2
here is the link for more information kindly sort it out or if it is any issue coz of me then let me know thanks
